analyses/cell-type-wilms-tumor-06/notebook_template/06_cnv_infercnv_exploration.Rmd

```diff
@@ -0,0 +1,445 @@
+
+#### Without reference
+
+![](`r file.path(result_dir, "reference-none_HMM-no" , glue::glue("06_infercnv_", params$sample_id, "_", "reference-none_HMM-no", "_heatmap.png"))`)
+
+#### With immune cells as reference
+
+![](`r file.path(result_dir, "reference-immune_HMM-no" , glue::glue("06_infercnv_", params$sample_id, "_", "reference-immune_HMM-no", "_heatmap.png"))`)
+
+#### With endothelium cells as reference
+
+![](`r file.path(result_dir, "reference-endothelium_HMM-no" , glue::glue("06_infercnv_", params$sample_id, "_", "reference-endothelium_HMM-no", "_heatmap.png"))`)
+
+#### With immune and endothelium cells as reference
+
+![](`r file.path(result_dir, "reference-both_HMM-no" , glue::glue("06_infercnv_", params$sample_id, "_", "reference-both_HMM-no", "_heatmap.png"))`)
+
+
+### Summarize CNV per chromosome and compartment
+
+```{r fig.width=20, fig.height=5, out.width='100%', out.height='100%'}
+infercnv_obj <- list()
+for(reference_value in c("reference-none", "reference-immune", "reference-endothelium", "reference-both")){
+  infercnv_obj[[reference_value]] <- readRDS(file.path(result_dir, glue::glue(reference_value, "_HMM-no") , glue::glue("06_infercnv_", params$sample_id, "_", reference_value, "_HMM-no.rds")))
+  print(Do_CNV_heatmap(object = srat, infercnv_obj = infercnv_obj[[reference_value]], group.by = "fetal_kidney_predicted.compartment"))
+}
+
+```
+
+### Try to calculate  single CNV score and see if it can be use to define cells with CNV
+
+```{r fig.width=20, fig.height=5, out.width='100%', out.height='100%', warning=FALSE}
+
+for(reference_value in c("reference-none", "reference-immune", "reference-endothelium", "reference-both")){
+  srat <- Do_CNV_score(srat,infercnv_obj = infercnv_obj[[reference_value]], reference_value)
+  p1 <- visualize_feature(srat, glue::glue("CNV-score_", reference_value) , group.by = "fetal_kidney_predicted.compartment" )
+  p2 <- visualize_density(srat, features=glue::glue("CNV-score_", reference_value), group.by = "fetal_kidney_predicted.compartment")
+  print(p1 + p2 + plot_layout(ncol = 3, widths = c(1,2,2)))
+}
+
+```
+
+This unique CNV score do not look so promising. 
+We might have to select chromosomes we would like to look at, i.e. the one relevant for Wilms tumor.
+
+
+### Explore infercnv results generated with immune and endothelial cells as reference, using a HMM i3 model
+
+#### Load the `Seurat` object
+
+We load the `Seurat` object generated in `06_infercnv.R`
+
+```{r}
+
+srat <- readRDS(file.path(module_base, "results", params$sample_id, glue::glue("06_infercnv_HMM-i3_", params$sample_id, "_reference-both.rds")))
+
+```
+
+#### Plot UMAP of known Wilms tumor related alteration
+
+As a reminder, we plot the label transfer from the fetal kidney atlas. 
+
+```{r fig.width=20, fig.height=8, out.width='100%'}
+
+visualize_metadata(srat, "seurat_clusters", "fetal_kidney_predicted.compartment")
```

Comment:
All the colors in this plot are very hard to see. Maybe it would make sense to flip the clusters and compartment labels, so that clusters are on the x axis and compartment is the colors?